src/expression_evaluator/node_rel_evaluator.cpp

```diff
@@ -9,6 +9,7 @@ namespace kuzu {
 namespace evaluator {
 
 void NodeRelExpressionEvaluator::evaluate() {
+    resultVector->resetAuxiliaryBuffer();
```

Comment:
Guess we should move resultVector->resetAuxiliaryBuffer to sourceOperator::getNextTuple. 
So each sourceOperator should reset auxiliaryBuffer beforing writing to the vector.

Reply:
As we discussed, this is not as trivial as reset on source operator. All expression evaluator should reset.